src/enumo/scheduler.rs

```diff
@@ -26,17 +27,19 @@ impl Scheduler {
 
     pub fn run<L: SynthLanguage>(
         &self,
-        egraph: EGraph<L, SynthAnalysis>,
+        egraph: &EGraph<L, SynthAnalysis>,
         ruleset: &Ruleset<L>,
-    ) -> Runner<L, SynthAnalysis> {
+    ) -> EGraph<L, SynthAnalysis> {
```

Comment:
Interface nitpick. Why an `EGraph` over a `Runner`? The runner has more information about the run. Maybe we don't need it though.

Reply:
Some runs have multiple phases + multiple runners. Could make sense to return the last runner, I guess, but that's kind of confusing because it seems like it implies that runner would know everything about the run, when it would actually just be for the last phase. I think returning an egraph doesn't have the same issue